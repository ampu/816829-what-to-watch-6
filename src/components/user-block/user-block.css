.user-block {
  position: relative;
  z-index: 1;
}
