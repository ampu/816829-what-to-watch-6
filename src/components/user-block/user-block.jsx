import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';

import {MainPath} from '../../constants/paths';
import {OperationStatus, OPERATION_STATUSES} from '../../constants/operation-status';

const UserBlock = ({loginStatus, user}) => {
  return (
    <div className="user-block">
      {loginStatus === OperationStatus.RESOLVED && (
        <div className="user-block__avatar">
          <Link to={MainPath.MY_LIST}>
            <img src={user.avatar} alt="User avatar" width="63" height="63"/>
          </Link>
        </div>
      )}

      {[OperationStatus.UNSET, OperationStatus.REJECTED].includes(loginStatus) && (
        <Link to={MainPath.SIGN_IN} className="user-block__link">Sign in</Link>
      )}
    </div>
  );
};

UserBlock.propTypes = {
  loginStatus: PropTypes.oneOf(OPERATION_STATUSES),
  user: PropTypes.shape({
    avatar: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }),
};

const mapStateToProps = (state) => ({
  loginStatus: state.loginStatus,
  user: state.user,
});

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
