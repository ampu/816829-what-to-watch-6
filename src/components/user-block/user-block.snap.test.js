import React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';

import {render} from '@testing-library/react';

import {UserBlock} from './user-block';
import OperationStatus from '../../constants/operation-status';

describe(`UserBlock component`, () => {
  it(`Should render according to the snapshot while authorization status is unknown`, () => {
    [
      undefined,
      OperationStatus.UNSET,
      OperationStatus.PENDING,
    ].forEach((loginStatus) => {
      const {container} = render(<UserBlock loginStatus={loginStatus}/>);
      expect(container).toMatchSnapshot();
    });
  });

  it(`Should render according to the snapshot while not logged in`, () => {
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <UserBlock loginStatus={OperationStatus.REJECTED}/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot while logged in`, () => {
    const user = {
      avatar: `fakeAvatar`,
    };
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <UserBlock loginStatus={OperationStatus.RESOLVED} user={user}/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });
});
