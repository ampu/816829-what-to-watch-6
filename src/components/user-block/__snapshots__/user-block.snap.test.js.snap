// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock component Should render according to the snapshot while authorization status is unknown 1`] = `
<div>
  <div
    class="user-block"
  />
</div>
`;

exports[`UserBlock component Should render according to the snapshot while authorization status is unknown 2`] = `
<div>
  <div
    class="user-block"
  />
</div>
`;

exports[`UserBlock component Should render according to the snapshot while authorization status is unknown 3`] = `
<div>
  <div
    class="user-block"
  />
</div>
`;

exports[`UserBlock component Should render according to the snapshot while logged in 1`] = `
<div>
  <div
    class="user-block"
  >
    <div
      class="user-block__avatar"
    >
      <a
        href="/mylist"
      >
        <img
          alt="User avatar"
          height="63"
          src="fakeAvatar"
          width="63"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`UserBlock component Should render according to the snapshot while not logged in 1`] = `
<div>
  <div
    class="user-block"
  >
    <a
      class="user-block__link"
      href="/login"
    >
      Sign in
    </a>
  </div>
</div>
`;
